lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -258,6 +259,27 @@ SILLocation tf::getUserSourceLocation(SILInstruction *inst) {
   return getUserSourceLocation(inst->getDebugLocation());
 }
 
+bool tf::isStatefulOp(const GraphOperationInst *graphOp) {
+  StringRef mangledName = graphOp->getName().str();
+  // Is this a known stateful op used in partitioning?
+  if (mangledName.startswith("tfc.SendToHost") ||
+      mangledName.startswith("tfc.RecvFromHost")) {
+    return true;
+  }
+  // Is this a stateful TensorFlow op?
+  std::unique_ptr<TF_Status, decltype(&TF_DeleteStatus)> status(
+    TF_NewStatus(), TF_DeleteStatus);
+  std::string opName = mangledName.substr(0, mangledName.find(','));
+  int isStateful = TF_OpIsStateful(opName.c_str(), status.get());
+  TF_Code statusCode = TF_GetCode(status.get());
+  // FIXME(SR-9388): we should actually fail if statusCode != TF_OK. However,
+  // the current tests use arbitrary strings in graph_op. Therefore, we will
+  // just allow TF_NOT_FOUND for now to avoid changing too many tests.
+  assert((statusCode == TF_OK || statusCode == TF_NOT_FOUND) &&
```

Comment:
can we enable the " || statusCode == TF_NOT_FOUND" under a test-only flag (the flag is always set, for all unit tests, and will not be retired)?

i feel pretty strongly that otherwise this can be hard to debug for user models.

Reply:
I'm fine either way. Thanks.